### PR TITLE
Feature/pre rules

### PR DESCRIPTION
**Pull request by dbanty**
For #142, adds a bunch of rules related to pre-releases for `dephell project bump`.  Also fixed some tests related to that function which weren't checking output correctly.

### CODE REVIEW

File: dephell/constants.py [hunk 1]
```diff
@@ -79,27 +79,38 @@ class ReturnCodes(Enum):
...
+PATCH_VERSIONS = VERSION_PATCH + VERSION_PRE_PATCH
+VERSION_RELEASE = ('release', )
 # semver has no post-releases: https://github.com/semver/semver/issues/200
 VERSION_POST = ('post', )
 VERSION_DEV = ('dev', )
 VERSION_LOCAL = ('local', )
 VERSION_SCHEMES = MappingProxyType(dict(
     # https://www.python.org/dev/peps/pep-0440/#version-scheme
-    pep=VERSION_MAJOR + VERSION_MINOR + VERSION_PATCH + VERSION_PRE + VERSION_POST + VERSION_DEV + VERSION_LOCAL,
+    pep=(
```
Comment (dbanty) [inline dephell/constants.py]: @orsinium this is going to fail flake8 (at least it did locally).  Can't have a binary operator immediately before or after a line break, but without line breaks the line is too long.  I can fake out flake8 by adding the + after the closing parens on line 99, but that is cheap.

Struggling with things like this is why I want an auto-formatter to save me time 😄.  What should I do?  Get more clever with how I bundle the version combos (like I did for `MAJOR_VERSIONS`)?

**Comment by orsinium** [comment]
Thank you! I'll check it today. Don't worry about flake, I can take care of it.

**Comment by orsinium** [comment]
Perfect! Now the function is much more complex, but it's my fault. I'll move it in the separated repo, I promise :)

I've fixed flake8. You can not worry about details like this. The main things that you're contributing are ideas and implementations.

**Comment by dbanty** [comment]
Thanks Gram!  Yes I felt bad about further complicating the function but didn't want to start chopping it up since you're going to move it anyway.

**Comment by orsinium** [comment]
I'll reach you for review in the new repository, if you don't mind.

**Comment by dbanty** [comment]
Of course, not a problem.

**Pull request merged**